### ISSUE TITLE
`promise/always-return` should not be reported when process exits

### ISSUE THREAD

**Issue by nandi95** [eslint-community/eslint-plugin-promise#451]
### Description

If I do `process.exit(0);` in a .then callback no return value should be expected from the callback as after the exit the rest of the code is unreachable.

### Steps to Reproduce

```ts
 void fetch(
     'my-url',
     {
         method: 'POST',
         headers: {
             // eslint-disable-next-line @typescript-eslint/naming-convention
             'Content-Type': 'application/json'
         },
         body: JSON.stringify({...})
     }
     // eslint-disable-next-line promise/always-return (this should not be needed)
 ).then(() => {
     task.stop();
     process.exit(0);
});
```

**Expected behavior:** not to report issue when process exists

**Actual behavior:** issue reported

### Versions

[Please fill this in if you are submitting a bug report]

- Node version: 21.6.1
- ESLint version: ^8.23.1
- eslint-plugin-promise version: ^6.0.0

**Comment by scagood** [comment]
I think you should use eslint-plugin-n with [process-exit-as-throw](https://github.com/eslint-community/eslint-plugin-n/blob/master/docs%2Frules%2Fprocess-exit-as-throw.md).
If you do then something probably is wrong 🤔

But, I am not sure that this should be in the promise plugin.

**Comment by brettz9** [comment]
I think it is reasonable to have an exception like this specific to Promise rules.
